react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx

```diff
@@ -53,30 +100,34 @@ export function RuleBasedOutputsSelector({ ruleSet }: ColorOverlayProps): ReactE
           .map((item) => item.assets)
           .flat() ?? [];
 
-      generateRuleBasedOutputs(model, contextualizedAssetNodes, flatMappings, ruleSet);
+      const basedNodeStyling = applyBasedNodeStyling(model, flatMappings);
+
+      const collectionStylings = await generateRuleBasedOutputs(
+        model,
+        contextualizedAssetNodes,
+        flatMappings,
+        ruleSet
+      );
+
+      const ruleNodeCollectionStylings: AssetStylingGroupAndStyleIndex[] = [];
+      for await (const styling of collectionStylings) {
```

Comment:
I didn't of this `for await`-syntax :eyes: Looks fancy, but what does it do in this case? It seems to me like `styling` is not a Promise type? Or is it?

Reply:
Yeah, it was returning a list of Promises so the "for away" is used to get the async "returnings" in a loop.
But I forgot that we dont use this pattern in here so I've changed to Promise.all to follow our internal pattern:
https://github.com/cognitedata/reveal/pull/4296/commits/553ce2a145f6f71b97d84ac9ee189bfed00381dc

Reply:
To be honest, I think this syntax looks better :sweat_smile: But we can perhaps transition to that later on